spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/RuntimeAppsController.java

```diff
@@ -81,6 +84,8 @@ public int compare(AppInstanceStatus i1, AppInstanceStatus i2) {
 
 	private final ResourceAssembler<AppStatus, AppStatusResource> statusAssembler = new Assembler();
 
+	private ForkJoinPool forkJoinPool = new ForkJoinPool(8);
```

Comment:
I'd recommend checking out `org.springframework.scheduling.concurrent.ForkJoinPoolFactoryBean`. As a library we should not just create threads which are beyond what user can control and that's why `ForkJoinPoolFactoryBean` allows user to set thread factory. Also we don't exactly handle any errors as that factory bean also have `uncaughtExceptionHandler`. Maybe this could be a shared component so that pool gets proper shutdown and possibly allows to complete running threads.

I can't say that I'm that familiar with `ForkJoinPool` but just throwing out default value `8` kinda smells a little.